src/PlexShareScreenshare/Server/ScreenStitcher.cs

```diff
@@ -0,0 +1,124 @@
+﻿
+///<author>Aditya Agarwal</author>
+///<summary> 
+///This file contains the ScreenStitcher Class that implements the
+///screen stitching functionality. It is used by ScreenshareServer. 
+///</summary>
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Controls;
+using static System.Net.Mime.MediaTypeNames;
+
+
+// Each frame consists of the resolution of the image and the ImageDiffList
+using Frame = System.Tuple<System.Tuple<int, int>,
+                        System.Collections.Generic.List<System.Tuple<System.Tuple<int, int>,
+                        System.Tuple<int, int, int>>>>;
+
+
+namespace ScreenShare.Server
+{
+    /// <summary>
+    /// Class contains implementation of the screen stitching using threads (tasks)
+    /// </summary>
+    internal class ScreenStitcher
+    {
+        private SharedClientScreen _sharedClientScreen;
+
+        private Task _stitchTask;
+        private Bitmap _stichedImage;
+
+        // new frame to hold the updated pixels
+        private Frame _newFrame;
+        private Bitmap _oldImage;
+        private Tuple<int, int> _resolution;
+
+        // Token and its source for killing the task
+        private CancellationTokenSource Source;
+        private CancellationToken Token;
+
+        // Called by the `SharedClientScreen`
+        ScreenStitcher(SharedClientScreen scs)
+        {
+            _stichedImage = null;
+            _newFrame = null;
+            _oldImage = null;
+            _stitchTask = null;
+            _resolution = null;
+        }
+        /// <summary>
+        /// Creates(if not exist) and start the task `_stitchTask`
+        /// Will read the image using `_sharedClientScreen.GetFrame`
+        /// and puts the final image using `_sharedClientScreen.PutFinalImage`
+        /// </summary>
+        public void StartStitching()
+        {
+            Source = new CancellationTokenSource();
+            Token = Source.Token;
+
+            if (_stitchTask == null)
+            {
+                while (true)
```

Comment:
The while should be a part of the  lambda function. The task is to be created just one time